docs: add comprehensive Meilisearch setup documentation

iamgerwin opened the pull request:
## Summary
- Created comprehensive documentation for Meilisearch setup and all scripts in the scripts/ directory
- Added platform-specific installation instructions for Windows, macOS, and Linux
- Updated CONTRIBUTING.md to reference the new documentation

## Changes Made

### New Documentation (`scripts/README.md`)
- **Getting Started with Meilisearch**: Step-by-step setup guide with platform-specific instructions
- **Working with Data**: Documentation for all data processing scripts
- **Building the Site**: Explanation of sitemap and llms.txt generation
- **Managing Contractors**: Guide for contractor data workflow  
- **Flood Control Projects**: Documentation for flood control data management
- **Troubleshooting**: Common issues and solutions

### Updated Files
- **CONTRIBUTING.md**: Added section referencing the new Meilisearch setup documentation

## Documentation Features
- Human-friendly, conversational tone
- Platform-specific installation guides (Windows, macOS, Linux, Docker)
- Clear step-by-step instructions
- Troubleshooting section for common issues
- Comprehensive script documentation

## Testing
- Verified all npm scripts mentioned in the documentation exist in package.json
- Confirmed all file paths are correct
- Documentation formatted with proper Markdown

Closes #289

codegino reviewed (approved):
LGTM

imjeffparedes reviewed (changes requested):
Consider moving to docs/

iamgerwin commented:
Hey Jeff! Thanks for the great feedback on the documentation placement. 

You're absolutely right that the scripts directory should contain code scripts and not documentation. I've moved the comprehensive Meilisearch setup documentation from `scripts/README.md` to `docs/Meilisearch.md` where it's more appropriately placed.

The changes include:
- Relocated the entire documentation to `docs/Meilisearch.md` 
- Updated all references in `CONTRIBUTING.md` to point to the new location
- Adjusted the title to better reflect that it covers both Meilisearch setup and scripts documentation

This should make the documentation easier to find and maintain a cleaner separation between code and documentation. Let me know if you have any other suggestions!

niculistana commented:
Thanks @iamgerwin I have no idea how to get the master key. Could we please add a documentation on how to get that if we have to ask on Discord or something else?

niculistana reviewed (dismissed):
Approved w/ some comments. @jasontorres @codegino feel free to chime-in if you have other thoughts. Thank you.

CC: @Sheape

Sheape commented:
LGTM for the most part.

I don't think the patch is necessary @niculistana for this case:
```diff
// some client-side code
const { searchClient } = instantMeiliSearch(
-  `${MEILISEARCH_HOST}:${MEILISEARCH_PORT}`,
+  `${MEILISEARCH_HOST}`,
  MEILISEARCH_SEARCH_API_KEY,
```

Works on my machine atleast with the following variables in my `.env` file:
```bash
VITE_MEILISEARCH_HOST=http://localhost
VITE_MEILISEARCH_PORT=7700
```
I agree we do need a `index:all` command, not to mention the script at `scripts/index_meilisearch.cjs` is also broken (I fixed it locally). I'll open a follow-up PR after this.

The Meilisearch master key can actually be any value ([in their docs](https://www.meilisearch.com/docs/learn/self_hosted/getting_started_with_self_hosted_meilisearch#running-meilisearch), as an example they set it as `./meilisearch --master-key="aSampleMasterKey"`). However, this should also be added in this docs.

iamgerwin commented:
Thank you all for the thorough review and valuable feedback! I have addressed all the suggestions:

## Changes implemented:

### Documentation improvements (@Sheape):
✅ Added `PERPLEXITY_API_KEY` requirement documentation for contractor commands with link to [Perplexity API settings](https://www.perplexity.ai/settings/api)
✅ Clarified that master key can be any value with example (`aSampleMasterKey`)
✅ Added reference to official Meilisearch docs for master key configuration

### Setup enhancements (@niculistana):
✅ Added comprehensive step-by-step setup instructions
✅ Included data population steps before key generation
✅ Added verification step to ensure proper setup
✅ Created `index:all` command in package.json for convenient batch indexing

### Script validation (@Sheape):
✅ Verified `index_meilisearch.cjs` script is working correctly - the connection error was due to Meilisearch not running, not a script issue

## Summary of commits:
- `2a2a86a`: Enhanced Meilisearch documentation with all reviewer feedback
- `f4b29c7`: Added `index:all` command for convenient batch indexing

The documentation now provides a clearer, more complete setup guide that addresses all the pain points mentioned in the review. The `index:all` command simplifies the setup process significantly for new developers.

Ready for re-review when you have a chance!

jasontorres commented:
Good stuff! Glad you figured it out without me onboarding you guys. 

```
"index:create-key": "node ./scripts/create_search_key.cjs", // this generates a read only search key we use for public use
"load:flood-control:arcgis": "node ./src/data/flood_control/load_flood_control_arcgis.js", // this can be deleted now. This is my failed attempt to load it to Cloudflare D1. Eventually I figured it's faster to use meilisearch and it worked out really well for us!
```

Sheape reviewed (approved):
Everything looks good to me. Thanks so much @iamgerwin.

Sheape commented on package.json:
```diff
@@ -26,6 +26,7 @@
     "index:create-key": "node ./scripts/create_search_key.cjs",
     "index:contractor-profiles": "node ./scripts/index_contractors.cjs",
+    "index:all": "npm run index:meilisearch && npm run index:flood-control:arcgis && npm run index:contractor-profiles",
     "load:flood-control:arcgis": "node ./src/data/flood_control/load_flood_control_arcgis.js",
```
```suggestion

```
We can delete this now as suggested by Jason.
